common/remote_object_test.go

```diff
@@ -0,0 +1,263 @@
+			Subtype:  "node",
+			ObjectID: remoteObjectID,
+			Preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					{Name: "num", Type: runtime.TypeNumber, Value: "1"},
+				},
+			},
+		}
+
+		val, err := valueFromRemoteObject(ctx, remoteObject)
+		require.NoError(t, err)
+		assert.Equal(t, rt.ToValue(map[string]interface{}{"num": float64(1)}), val)
+	})
+
+}
+
+func TestParseRemoteObject(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name     string
+		preview  *runtime.ObjectPreview
+		value    string
+		expected map[string]interface{}
+		expErr   interface{}
+	}{
+		{
+			name: "most_types",
+			preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					{Name: "accessor", Type: runtime.TypeAccessor, Value: ""},
+					{Name: "bigint", Type: runtime.TypeBigint, Value: "100n"},
+					{Name: "bool", Type: runtime.TypeBoolean, Value: "true"},
+					{Name: "fn", Type: runtime.TypeFunction, Value: ""},
+					{Name: "num", Type: runtime.TypeNumber, Value: "1"},
+					{Name: "str", Type: runtime.TypeString, Value: "string"},
+					{Name: "strquot", Type: runtime.TypeString, Value: `"quoted string"`},
+					{Name: "sym", Type: runtime.TypeSymbol, Value: "Symbol()"},
+				},
+			},
+			expected: map[string]interface{}{
+				"accessor": "accessor",
+				"bigint":   int64(100),
+				"bool":     true,
+				"fn":       "function()",
+				"num":      float64(1),
+				"str":      "string",
+				"strquot":  "quoted string",
+				"sym":      "Symbol()",
+			},
+		},
+		{
+			name: "nested",
+			preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					// We don't actually get nested ObjectPreviews from CDP.
+					// I.e. the object `{nested: {one: 1}}` returns value "Object"
+					// for the "nested" property, with a nil *ValuePreview. :-/
+					{Name: "nested", Type: runtime.TypeObject, Value: "Object"},
+				},
+			},
+			expected: map[string]interface{}{
+				"nested": "Object",
+			},
+		},
+		{
+			name:     "overflow",
+			preview:  &runtime.ObjectPreview{Overflow: true},
+			expected: map[string]interface{}{},
+			expErr:   &objectOverflowError{},
+		},
+		{
+			name: "err_parsing_property",
+			preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					{Name: "failprop", Type: runtime.TypeObject, Value: "some"},
+				},
+			},
+			expected: map[string]interface{}{},
+			expErr:   &objectPropertyParseError{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			remoteObject := &runtime.RemoteObject{
+				Type:     "object",
+				ObjectID: runtime.RemoteObjectID("object_id_0123456789"),
+				Preview:  tc.preview,
+				Value:    easyjson.RawMessage(tc.value),
+			}
+			val, err := parseRemoteObject(remoteObject)
+			assert.Equal(t, tc.expected, val)
+			if tc.expErr != nil {
+				// FIXME: This passes regardless of the error set in tc.expErr :-S
+				assert.ErrorAs(t, err, &tc.expErr)
```

Comment:
I wasn't able to make this work :confused: This would pass even if `expErr: errors.New("")`... It doesn't seem related to `multierror`, it would happen even if I unwrapped the internal error. It's something with `errors.As()` and how the type is specified in the struct, I couldn't figure it out.

Reply:
I suggest checking for error text instead of trying to match the exact error type as it ties to the implementation.

Reply:
Added my suggestions :)

Reply:
In general I'd disagree with that; checking the error message is brittle and not much better than checking the log message itself... But since it would avoid this type checking issue, sure, I'll change it.

Let's just merge this sometime today, we're bikeshedding at this point.

Reply:
Yeah, in general, I agree too. However, this is a mini parser code, so it's ok here. Log text checking is a different world in itself 😅

Reply:
Resolved by 6c57782.